server/methods.coffee

```diff
@@ -11,17 +12,30 @@ Meteor.methods
     @LegTotals.find().count()
 
   getArticleCounts: =>
-    console.log 'inside article method call', @ArticleTotals
-    @ArticleTotals
+    @ArticleCounts.find().count()
+
+  getDiseaseInfo: (disease) =>
+    diseaseInfo = @ArticleCounts.aggregate([
+        {$match: {"subject.diseaseLabels": disease}},
+        {$project : { day : {$substr: ["$promedDate", 0, 10] }}},
+        {$group   : { _id : "$day",  number : { $sum : 1 }}},
+        {$sort    : { _id : 1 } }
+    ])
+    diseaseInfo
+
+  getDiseaseNames: =>
+    if !diseases
+      diseases = @ArticleCounts.find({},
+                    fields: 'subject.diseaseLabels': true
+                  ).fetch().map((x) ->
+                    x.subject.diseaseLabels
+                  )
+      diseases = _.flatten(diseases)
+      diseases = _.uniq(diseases).sort()
```

Comment:
I'd merge the last two assignments into one nested function call.
Re-assigning arrays is sort of a strange operation to perform, since they're considered objects in JS, but being treated as chunks of data that get re-created and moved, unlike just re-assigning pointers as it is the case with objects.